File: src/app/Class/News.ts
import { SafeUrl, SafeResourceUrl } from '@angular/platform-browser';

export class News {

    id: number;
    newsTitle: string;
    newsCreationDate: Date;
    creator: string;
    newsContent: string;
    newsImage: SafeResourceUrl;

    constructor(obj?: any) {
        this.id = (obj && obj.id) || 0;
        this.newsTitle = (obj && obj.newsTitle) || '';
        this.newsCreationDate = (obj && obj.newsCreationDate) || null;
        this.creator = (obj && obj.creator) || '';
        this.newsContent = (obj && obj.contentNewsDirFile) || '';
        this.newsImage = (obj && obj.fotoNewsDirFile) || '';
    }

    // Getters

    getId(): number {
        return this.id;
    }

    getNewsTitle(): string {
        return this.newsTitle;
    }

    getNewsCreationDate(): Date {
        return this.newsCreationDate;
    }

    getCreator(): string {
        return this.creator;
    }

    getNewsContent(): string {
        return this.newsContent;
    }

    getNewsImage(): SafeResourceUrl {
        return this.newsImage;
    }

    // Setters

    setId(id: number) {
        this.id = id;
    }

    setNewsTitle(newsTitle: string) {
        this.newsTitle = newsTitle;
    }

    setNewsCreationDate(creationDate: number) {
        this.newsCreationDate = new Date(creationDate);
    }

    setCreator(creator: string) {
        this.creator = creator;
    }

    setNewsContent(newsContent: string) {
        this.newsContent = newsContent;
    }

    setNewsImage(newsImage: SafeResourceUrl) {
        this.newsImage = newsImage;
    }

}

File: Backend/src/main/java/musiccube/entities/News.java
package musiccube.entities;


import javax.persistence.*;
import java.util.Date;

@Entity
public class News {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @Column
    private String newsTitle;

    @Column
    private Date newsCreationDate;

    @Column
    private String creator;

    @Column
    private String contentNewsDirFile;

    @Column
    private String imageUrl;

    public News() {
    }

    public News(String newsTitle, Date newsCreationDate, String creator, String contentNewsDirFile) {
        this.newsTitle = newsTitle;
        this.newsCreationDate = newsCreationDate;
        this.creator = creator;
        this.contentNewsDirFile = contentNewsDirFile;
    }

    public News(String newsTitle, Date newsCreationDate, String creator, String contentNewsDirFile, String imageUrl) {
        this.newsTitle = newsTitle;
        this.newsCreationDate = newsCreationDate;
        this.creator = creator;
        this.contentNewsDirFile = contentNewsDirFile;
        this.imageUrl = imageUrl;
    }

    public int getId() {
        return id;
    }

    public String getNewsTitle() {
        return newsTitle;
    }

    public void setNewsTitle(String newsTitle) {
        this.newsTitle = newsTitle;
    }

    public Date getNewsCreationDate() {
        return newsCreationDate;
    }

    public void setNewsCreationDate(Date newsCreationDate) {
        this.newsCreationDate = newsCreationDate;
    }

    public String getCreator() {
        return creator;
    }

    public void setCreator(String creator) {
        this.creator = creator;
    }

    public String getContentNewsDirFile() {
        return contentNewsDirFile;
    }

    public void setContentNewsDirFile(String contentNewsDirFile) {
        this.contentNewsDirFile = contentNewsDirFile;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
}
